04_documentation/code_documentation_tools_exercise.md

```diff
@@ -0,0 +1,92 @@
+
+  **Note** If you are using MkDocs inside a container, you have to [forward/publish the ports to your Host](https://docs.docker.com/config/containers/container-networking/).
+- Adapt the project documentation such that it the starting/index page contains information about the project. Open the `index.md` file inside the `docs` directory and replace its content with the content of the `README.md`. In contrast to Sphinx, we cannot include the `README.md` from the repository in the documentation.
+- [Configure the website](https://www.mkdocs.org/user-guide/configuration/) by editing the `mkdocs.yml` file. Changes made to this file might not been picked up by the interactive rebuilding of `mkdocs serve`. In this case, you have to stop `mkdocs serve` command (CTRL+C) and start it again afterwards.
+    - Change the name of the website (`site_name`) from "My Docs" to "SSE Documentation Example"
+    - [Add a navigation section](https://www.mkdocs.org/user-guide/writing-your-docs/#configure-pages-and-navigation) `nav` and add the three pages manually:
+        - Add the `index.md` with label `Home`.
+        - Create a section `User Guide` and add the following subsections
+            - Add the `compilation-and-installation.md` with label `Installation`.
+            - Add the `how-to.md` with label `Getting Started`
+          You should only have two buttons "Home" and "User Guide" in the generated website.
+    - Add the [repository URL](https://www.mkdocs.org/user-guide/configuration/#repo_url). This adds a "Edit on GitHub" button at the top right of the website. Try if this works. If it does not work, this is because older versions of MkDocs set a link to the `master` branch by default, but GitHub calls this branch `main` nowadays. In order to point to the correct branch, set the `edit_uri` setting to `edit/main/docs/`.
+
+        **Note** The "Edit on GitHub" also fails if you have not pushed the corresponding Markdown file to your repository yet.
+
+    - Make yourself the [website's author](https://www.mkdocs.org/user-guide/configuration/#site_author). You can use your real name or your GitHub username. This setting will not change the appearance of your website, but adds the name to the HTML metadata.
+- [Deploy your website](https://www.mkdocs.org/user-guide/deploying-your-docs/) to [GitHub Pages](https://pages.github.com/). This consists of following steps:
+    - Run `mkdocs gh-deploy` to build and deploy the website:
+
+      Deployment of the website will create a branch `gh-pages` inside the repository if it does not already exist. This branch contains all files needed to deliver the website. You should not edit any of the files in the branch manually since `mkdocs gh-deploy` will overwrite all changes. Instead, fix problems in your main branch and deploy the website afterwards again.
```

Comment:
```suggestion
      Deployment of the website creates a branch `gh-pages` inside the repository if it does not already exist and pushes it to the default remote. The branch contains all files needed to deliver the website. You should not edit any of the files in the branch manually since `mkdocs gh-deploy` overwrites all changes. Instead, always fix problems in the main branch and deploy the website afterwards again.
```